File: Python/Algorithm/Floyd_Warshall.py
# Complexity: O(n^3)
# Use: Used to determine the shortest distance of all connected nodes

def fw(vr, eg):
    dist = [[float('inf') for _ in range(vr)] for _ in range(vr)]
    for i in range(vr): dist[i][i] = 0
    for v1, v2, weight in eg: dist[v1-1][v2-1] = weight
    for k in range(vr):
        for i in range(vr):
            for j in range(vr):
                dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
    return dist

# Example usage
vr = int(input("Enter number of vr: "))
eg = []
print("Enter eg in the format: v1 v2 weight (Enter 'done' when finished):")
while True:
    edge = input()
    if edge == 'done': break
    v1, v2, weight = map(int, edge.split())
    eg.append((v1, v2, weight))
dis = fw(vr, eg)

# Printing the shortest dis
for i in range(vr):
    for j in range(vr):
        print(-1 if dis[i][j] == float('inf') else dis[i][j], end=' ')
    print()

File: CPP/Algorithm/Floyd_Warshall.cpp
// Complexity: O(n^3)
// Use: Used to determine the shortest distance of all connected nodes

long long dis[N][N];
void pre(int vr) {
    for (int k = 1; k <= vr; ++k) {
        for (int i = 1; i <= vr; ++i) {
            for (int j = 1; j <= vr; ++j) {
                dis[i][j] = (i == j ? 0 : min(dis[i][j], dis[i][k] + dis[k][j]));
            }
        }
    }
}

// Operation:
    int vr, eg; cin >> vr >> eg;
    fill(&dis[0][0], &dis[0][0] + (N * N), LLONG_MAX);
    for(i = 0; i < eg; ++i) {
        long long v1, v2, wt; cin >> v1 >> v2 >> wt;
        dis[v1][v2] = wt;
        // g[v2].emplace_back(v1, wt); // In case of undirected graph
    }
    pre(vr);
    for(i = 1; i <= vr; ++i) {
        for(j = 1; j <= vr; ++j)
            cout << (dis[i][j] == INT_MAX ? -1 : dis[i][j]) << ' '; cout << '\n';
    }

// OR,

#include<bits/stdc++.h>
using namespace std;

const int N = 105;
int d[N][N];
int main() {
  int n = 10;
  for (int i = 1; i <= n; i++) {
    for (int j = 1; j <= n; j++) {
      if (i != j) {
        d[i][j] = 1e9;
      }
    }
  }
  for (int k = 1; k <= n; ++k) {
    for (int i = 1; i <= n; ++i) {
      for (int j = 1; j <= n; ++j) {
        d[i][j] = min(d[i][j], d[i][k] + d[k][j]);
      }
    }
  }
  return 0;
}